test/e2e/app-dir/next-form/default/pages/pages-dir/forms/with-function/action-client/index.tsx

```diff
@@ -0,0 +1,33 @@
+'use client'
+import * as React from 'react'
+import { useActionState, useState } from 'react'
+import Form from 'next/form'
+import { useRouter } from 'next/router'
+
+export default function Page() {
+  const destination = '/pages-dir/redirected-from-action'
+  const router = useRouter()
+  const [, dispatch] = useActionState(() => {
```

Comment:
Can we make these tests work in React 18 or is Form not even supported in React 18?

The tests should be gated then e.g. `const isReact18 = parseInt(process.env.NEXT_TEST_REACT_VERSION) === 18`.

Reply:
yeah sure, good catch!